pkg/cmd/pr/create/regex_writer.go

```diff
@@ -0,0 +1,62 @@
+package create
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"regexp"
+)
+
+func NewRegexWriter(out io.Writer, regexp *regexp.Regexp, repl string) io.Writer {
+	return &RegexWriter{out: out, regexp: *regexp, repl: repl}
+}
+
+type RegexWriter struct {
+	out    io.Writer
+	regexp regexp.Regexp
+	repl   string
+}
+
+func (s RegexWriter) Write(data []byte) (int, error) {
+	filtered := []byte{}
+	repl := []byte(s.repl)
+	scanner := bufio.NewScanner(bytes.NewReader(data))
+	scanner.Split(scanLines)
+
+	for scanner.Scan() {
+		b := scanner.Bytes()
+		f := s.regexp.ReplaceAll(b, repl)
+		if len(f) > 0 {
+			filtered = append(filtered, f...)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return 0, err
+	}
+
+	if len(filtered) != 0 {
+		_, err := s.out.Write(filtered)
+		if err != nil {
+			return 0, err
+		}
+	}
+
+	return len(data), nil
+}
+
+func scanLines(data []byte, atEOF bool) (advance int, token []byte, err error) {
```

Comment:
The only difference between this and `bufio.ScanLines` seems to be that this doesn't drop trailing `\r` characters. Is that intentional? If not, you should probably consider removing this function and the call to `scanner.Split` (as `bufio.ScanLines` is the default)

Reply:
I was using the default `buffo.ScanLines` to begin with but that function also strips the token that it splits the lines at, in our case it is `\n`, which is why I wrote this one, to preserve them. It is a subtle change but on line 54 I am doing `i + 1`, instead of the just `i`.

Reply:
Interesting, I'm still not sure how that's useful: You can presume that a line ended in a newline (or was the end of the file, and I'm not sure you need to know the difference) because it comes from the scanner, and your regexp even matches against an optional newline char. It would make this much simpler to use the default SplitFunc and append a newline when writing to `out` (and maybe adding an extra newline at the end of the output in the worst case).

If you have hard requirements to use this version, I'd certainly recommend documenting why you have this and the subtle difference with the default - without that you might find someone doing precisely what I did and not noticing the difference before 'refactoring' it

Reply:
Agreed on documentation! 

I think this is still necessary to use this custom function due to your other comment https://github.com/cli/cli/pull/2016#discussion_r503196407 where the call to `Write()` is not guaranteed to end in a new line. 

Here is example I think that helps illustrate it better. We can ignore the regex replacement portion for the moment as it just adds unnecessary complexity for this example.

Line to be outputted from command: 
`line from command\n`

Pseudo-code calls to `Write()`:
`Write("line from")`
`Write("command\n")`

Output if we append `\n` when writing to `out`:
```
line from
command
```

Desired output:
```
line from command
```

We end up with an erroneous line break which is undesirable and potentially confusing.

Reply:
But if you use a scanner hung off the struct as discussed elsewhere it's buffering the content and will only return a complete line to the goroutine catching those

Reply:
Yup! 👍 For sure, in that scenario the default `scanLines` would work properly. If we move that direction I will make the switch.